spec/requests/api/v1/companies_spec.rb

```diff
@@ -0,0 +1,134 @@
+          expect(json_response).not_to include(*unexpected_data)
+        end
+      end
+
+      describe "filtering by industry" do
+        let(:industry) { companies.map(&:industry).sample }
+        let(:selected) { companies.select { |company| company.industry == industry } }
+        let(:unselected) { companies.reject { |company| company.industry == industry } }
+
+        before { get api_v1_companies_path, params: {query: {industry:}} }
+
+        let(:expected_data) do
+          selected.map do |company|
+            a_hash_including(
+              "id" => company.id,
+              "name" => company.name,
+              "employee_count" => company.employee_count,
+              "industry" => company.industry
+            )
+          end
+        end
+
+        let(:unexpected_data) do
+          unselected.map do |company|
+            a_hash_including(
+              "id" => company.id,
+              "name" => company.name,
+              "employee_count" => company.employee_count,
+              "industry" => company.industry
+            )
+          end
+        end
+
+        let(:json_response) { JSON.parse(response.body) }
+
+        it "returns the companies that match the industry" do
+          expect(json_response).to include(*expected_data)
+        end
+
+        it "does not return companies that do not match the industry" do
+          expect(json_response).not_to include(*unexpected_data)
+        end
+      end
+
+      describe "filtering by minimum employee count" do
+        let(:threshold) { companies.map(&:employee_count).sample }
+        let(:selected) { companies.select { |company| company.employee_count >= threshold } }
+        let(:unselected) { companies.reject { |company| company.employee_count >= threshold } }
+
+        before { get api_v1_companies_path, params: {query: {employee_count: threshold}} }
+
+        let(:expected_data) do
+          selected.map do |company|
+            a_hash_including(
+              "id" => company.id,
+              "name" => company.name,
+              "employee_count" => company.employee_count,
+              "industry" => company.industry
+            )
+          end
+        end
+
+        let(:unexpected_data) do
+          unselected.map do |company|
+            a_hash_including(
+              "id" => company.id,
+              "name" => company.name,
+              "employee_count" => company.employee_count,
+              "industry" => company.industry
+            )
+          end
+        end
+
+        let(:json_response) { JSON.parse(response.body) }
+
+        it "returns the companies with an employee count higher or equal to the param" do
+          expect(json_response).to include(*expected_data)
+        end
+
+        it "does not return companies with an employee count lower than the param" do
+          expect(json_response).not_to include(*unexpected_data)
+        end
+      end
+    end
+  end
+end
```

Comment:
At the moment of doing this Pull Request, I'm skipping two important scenarios:
Test for combined filter params and test for the total_deal_amount param. Both tests require a bit of thinking to make a good test (e.g. If I filter by a combined set of params how do I know the collection of filters narrowed down the search or if it was just one of them [name throws an exact match]).
However, I did manual test with postman before submitting. I'm worried about time